doajtest/unit/test_fc_publisher_update_request.py

```diff
@@ -0,0 +1,174 @@
+        assert fc.form is not None
+        assert fc.source is not None
+        assert fc.form_data is not None
+
+        # test each of the workflow components individually ...
+
+        # pre-validate and ensure that the disbaled fields get re-set
+        fc.pre_validate()
+        assert fc.form.pissn.data == "1234-5678"
+        assert fc.form.eissn.data == "9876-5432"
+        assert fc.form.contact_name.data == "Contact Name"
+        assert fc.form.contact_email.data == "contact@email.com"
+        assert fc.form.confirm_contact_email.data == "contact@email.com"
+
+        # run the validation itself
+        assert fc.validate(), fc.form.errors
+
+        # run the crosswalk (no need to look in detail, xwalks are tested elsewhere)
+        fc.form2target()
+        assert fc.target is not None
+
+        # patch the target with data from the source
+        fc.patch_target()
+        assert fc.target.created_date == "2000-01-01T00:00:00Z"
+        assert fc.target.id == "abcdefghijk"
+        assert len(fc.target.notes) == 2
+        assert fc.target.owner == "Owner"
+        assert fc.target.editor_group == "editorgroup"
+        assert fc.target.editor == "associate"
+        assert fc.target.application_status == constants.APPLICATION_STATUS_UPDATE_REQUEST # because it hasn't been finalised yet
+        assert fc.target.suggester['name'] == fc.source.suggester["name"]
+        assert fc.target.suggester['email'] == fc.source.suggester["email"]
+        assert fc.target.bibjson().replaces == ["1111-1111"]
+        assert fc.target.bibjson().is_replaced_by == ["2222-2222"]
+        assert fc.target.bibjson().discontinued_date == "2001-01-01"
+        assert fc.target.current_journal == "123456789987654321"
+        assert fc.target.related_journal == "987654321123456789"
+        assert fc.target.bibjson().subjects() == fc.source.bibjson().subjects()
+
+        # now do finalise (which will also re-run all of the steps above)
+        fc.finalise()
+        assert fc.target.application_status == constants.APPLICATION_STATUS_UPDATE_REQUEST
+
+        j2 = models.Journal.pull(journal.id)
+        assert j2.current_application == fc.target.id
+        assert fc.target.current_journal == j2.id
+
+    def test_02_conditional_disabled(self):
+        s = models.Suggestion(**deepcopy(UPDATE_REQUEST_SOURCE))
+
+        # source only, all fields disabled
+        fc = formcontext.ApplicationFormFactory.get_form_context(role="publisher", source=s)
+        assert "pissn" in fc.renderer.disabled_fields
+        assert "eissn" in fc.renderer.disabled_fields
+        assert "contact_name" in fc.renderer.disabled_fields
+        assert "contact_email" in fc.renderer.disabled_fields
+        assert "confirm_contact_email" in fc.renderer.disabled_fields
+        # should validate
+        assert fc.validate()
+
+        # source only, no contact details, so those fields not disabled
```

Comment:
The rules for this form (inherited from the reapplication form) are that:

* title, alternative_title, pissn and eissn are always disabled, the publisher is not allowed to modify them
* contact_email and contact_name are disabled if they are set, so that they can't be modified, but if they are not set they can be added.